### ISSUE TITLE
LVM logical volume with custom mountpoint is broken

### ISSUE THREAD

**Issue by sveldhuisen** [SUSE/kiwi#904]
## Problem description
I'm using the following LVM setup inside my config.xml for a VMX image based on SLES 12 SP4:

            <systemdisk name="systemVG">
                <volume name="swap" size="8G"/>
                <volume mountpoint="var" name="LVvar" size="115G"/>
                <volume mountpoint="var/opt" name="LVvar_opt" size="10G"/>
            </systemdisk>
            <size unit="G" additive="false">150</size>:

I use this to create a temporary logical volume which I convert to a swap partition on first boot. With Kiwi 7.x this was working fine. I needed to upgrade to Kiwi 9.15.3 due to SLES SP4 being applied. With the same Kiwi config I now get an error doing the build:

[ INFO    ]: 17:03:04 | Creating system image
[ INFO    ]: 17:03:04 | Preparing boot system
[ INFO    ]: 17:03:04 | Creating .profile environment
[ INFO    ]: 17:03:05 | Precalculating required disk size
[ INFO    ]: 17:03:06 | --> system data with filesystem overhead needs 8154 MB
[ INFO    ]: 17:03:06 | --> LVM overhead adding 80 MB
[ ERROR   ]: 17:03:06 | Unexpected error:
Traceback (most recent call last):
  File "/usr/bin/kiwi-ng", line 9, in <module>
    load_entry_point('kiwi==9.17.1', 'console_scripts', 'kiwi-ng-2')()
  File "/usr/lib/python2.7/site-packages/kiwi/kiwi.py", line 64, in main
    App()
  File "/usr/lib/python2.7/site-packages/kiwi/app.py", line 35, in __init__
    task_class().process()
  File "/usr/lib/python2.7/site-packages/kiwi/tasks/system_build.py", line 269, in process
    result = image_builder.create()
  File "/usr/lib/python2.7/site-packages/kiwi/builder/disk.py", line 170, in create
    result = disk.create_disk()
  File "/usr/lib/python2.7/site-packages/kiwi/builder/disk.py", line 221, in create_disk
    disksize_mbytes = self.disk_setup.get_disksize_mbytes()
  File "/usr/lib/python2.7/site-packages/kiwi/storage/setup.py", line 88, in get_disksize_mbytes
    root_filesystem_mbytes
  File "/usr/lib/python2.7/site-packages/kiwi/storage/setup.py", line 294, in _accumulate_volume_size
    data_volume_mbytes.volume[volume.realpath]
KeyError: 'swap' 

## Expected behaviour
A logical volume LVswap should be created with /swap as mountpoint. Documentation about the name attribute on the volume element states:

A required attribute. The name of the volume. If mountpoint is not specified, a directory with the given name will be created by KIWI if it does not already exist inside the root tree. However, if the name contains the KIWI internal path field separator “_”, it is required to specify the path in an additional mount point attribute. The special value @root can be combined with the size or freespace attribute to control the size of the root volume.

This seems to be broken in Kiwi 9.x

## Steps to reproduce the behaviour
Create a Logical volume with a custom mount point

## OS and Software information
+ KIWI version: 9.15.3 
+ Operating system: SLES 12 SP4
+ OBS version: none

**Comment by sveldhuisen** [comment]
My current workaround is creating the /swap directory inside the virtual root. This seems to work.

**Comment by davidcassany** [comment]
@sveldhuisen I could not reproduce this issue. I am using 

```xml
<systemdisk name="systemVG">
    <volume name="swap" size="1G"/>
    <volume mountpoint="var" name="LVvar" size="4G"/>
    <volume mountpoint="var/opt" name="LVvar_opt" size="2G"/>
</systemdisk>
```

Tried it with several kiwi versions, 9.15.3, master branch and 9.17.1 (from the logs this looks like the one you might be using). Could you provide/confirm some more information? I'll try to set a very close env to what you have.

* Which is building host environment? Is it SLE-12-SP4?
* Which is the filesystem of your host?
* Is your host using LVM?
* Are you using the kiwi provided from the official SLE12 repo?

**Comment by sveldhuisen** [comment]
@davidcassany that is a pity. I can share my full config.xml if you need it.

Building host environment: SLES 12 SP4
Volume manger: LVM2 with one volume group named "system"
filesystems are configured for EXT4

Kiwi version: I'm using 9.16.18 from the official SLES repositories at the moment. I can also reproduce it with 9.17.1 from http://download.opensuse.org/repositories/Virtualization:/Appliances:/Builder/SLE_12_SP3/x86_64/

**Comment by davidcassany** [comment]
Thanks for the details, I'll have a look again

**Comment by sveldhuisen** [comment]
Could it be, that the name of the Volume Group of the build machine ("system") interferes with the name of the Volume Group of the image (systemVG)?

**Comment by davidcassany** [comment]
> Could it be, that the name of the Volume Group of the build machine ("system") interferes with the name of the Volume Group of the image (systemVG)?

I am still unable to reproduce this issue. I set up a SP4 VM with a LVM volumes for `root` and `home`. On an ext4 filesystem. The volume group of my system is called `systemVG`, the default kiwi name. I used the published kiwi version (v9.16.18).

I tried building the attached images with `system` and `systemVG2nd` as volume group name, always without issues, so I doubt this is an issue of having the host being named `system` and using `systemVG` for the build image.

Can you give a try to the attached project? or check any relevant difference with yours?
[test.tar.gz](https://github.com/SUSE/kiwi/files/2689863/test.tar.gz)
